TLM/TLM/UI/Helpers/GuideWrapper.cs

```diff
@@ -0,0 +1,43 @@
+namespace TrafficManager.UI {
+    using System;
+    using CSUtil.Commons;
+
+    internal class GuideWrapper {
+        private GenericGuide m_guide;
+        internal GuideInfo m_info;
+
+        public GuideWrapper(string key) {
+            m_guide = new GenericGuide();
+            m_info = new GuideInfo {
+                // These values are game defaults:
+                m_delayType = GuideInfo.Delay.OccurrenceCount,
+                m_displayDelay = 1,
+                m_repeatDelay = 3, 
+                m_overrideOptions = true,
+                m_icon = "ToolbarIconZoomOutGlobe",
+                m_tag = "Generic",
+                m_name = key,
+            };
+        }
+
+        private bool CheckStack() {
+            if (!Environment.StackTrace.Contains("SimulationStep")) {
+                Log.Error("Guide should be handled from simulation step. stack is:\n" + Environment.StackTrace);
```

Comment:
Doesn't `Log.Error()` already output a stack trace?

Reply:
Yes it is :)